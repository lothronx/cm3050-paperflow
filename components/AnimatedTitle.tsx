"use client";

import { useEffect } from "react";
import { StyleSheet, View } from "react-native";
import Animated, { useSharedValue, useAnimatedStyle, withSpring } from "react-native-reanimated";
import { COLORS } from "@/constants/Colors";
import { Text } from "@/components/Text";

export const AnimatedTitle = () => {
  const opacity = useSharedValue(0);
  const translateY = useSharedValue(20);

  useEffect(() => {
    opacity.value = withSpring(1);
    translateY.value = withSpring(0);
  }, [opacity, translateY]);

  const animatedStyle = useAnimatedStyle(() => ({
    opacity: opacity.value,
    transform: [{ translateY: translateY.value }],
  }));

  return (
    <Animated.View style={[styles.container, animatedStyle]}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>Paper</Text>
        <Text style={[styles.title, styles.titleHighlight]}>Flow</Text>
      </View>
      <Text style={styles.subtitle}>Split Long Image for Printing</Text>
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    marginVertical: 40,
  },
  titleContainer: {
    flexDirection: "row",
    marginBottom: 8,
  },
  title: {
    fontSize: 48,
    fontWeight: "900",
    fontStyle: "italic",
    color: COLORS.text,
    fontFamily: "Playfair",
    textShadowColor: "rgba(0, 0, 0, 0.25)",
    textShadowOffset: {
      width: 0,
      height: 2,
    },
    textShadowRadius: 3.84,
  },
  titleHighlight: {
    color: COLORS.primary,
  },
  subtitle: {
    fontSize: 20,
    fontWeight: "400",
    color: COLORS.textSecondary,
    textShadowColor: "rgba(0, 0, 0, 0.25)",
    textShadowOffset: {
      width: 0,
      height: 2,
    },
    textShadowRadius: 3.84,
  },
});
