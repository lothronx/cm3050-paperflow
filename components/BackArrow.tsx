import { StyleSheet, View, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { COLORS } from "@/constants/Colors";
import { router } from "expo-router";

export const BackArrow = () => {
  return (
    <View style={styles.backArrow}>
      <TouchableOpacity onPress={() => router.back()}>
        <Ionicons name="chevron-back" size={24} color={COLORS.primary} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  backArrow: {
    position: "absolute",
    top: 64,
    left: 18,
    zIndex: 10,
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 999,
    backgroundColor: COLORS.background,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
