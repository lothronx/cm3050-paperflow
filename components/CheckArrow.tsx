/**
 * Check Arrow Component
 * 
 * Provides a confirmation button to proceed to the next step
 * Features:
 * - Checkmark icon
 * - Primary color background
 * - Shadow effect for visibility
 * - Absolute positioning in top-right corner
 */

// React Native core imports
import { StyleSheet, View, TouchableOpacity } from "react-native";

// Icon library
import { Ionicons } from "@expo/vector-icons";

// Custom constants
import { COLORS } from "@/constants/Colors";
import { MARGINS } from "@/constants/Margins";

/**
 * Props for CheckArrow component
 * 
 * @property onClick - Callback function when the check arrow is pressed
 */
type CheckArrowProps = {
  onClick: () => void;
};

/**
 * Check Arrow Component
 * 
 * Displays a confirmation button in the top-right corner of the screen
 */
export const CheckArrow = ({ onClick }: CheckArrowProps) => {
  return (
    <View style={styles.checkArrow}>
      <TouchableOpacity onPress={onClick} testID="check-arrow-button">
        <Ionicons name="checkmark" size={24} color={COLORS.background} />
      </TouchableOpacity>
    </View>
  );
};

// Styles for component layout and appearance
const styles = StyleSheet.create({
  checkArrow: {
    position: "absolute",
    top: MARGINS.top,
    right: MARGINS.horizontal,
    zIndex: 10,
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 999,
    backgroundColor: COLORS.primary,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
