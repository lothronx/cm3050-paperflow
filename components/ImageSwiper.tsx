/**
 * Image Swiper Component
 * 
 * Provides a image carousel with:
 * - Custom pagination dots
 * - Image carousel display
 * - Shadow effects for container
 */

// Swiper library for image carousel
import Swiper from "react-native-swiper";

// React Native core imports
import { View, Image, StyleSheet } from "react-native";

// Custom constants
import { COLORS } from "@/constants/Colors";
import { MARGINS } from "@/constants/Margins";

/**
 * Props for ImageSwiper component
 * 
 * @property images - Array of image URIs to display in the swiper
 */
interface ImageSwiperProps {
  images: string[];
}

/**
 * Image Swiper Component
 * 
 * Displays a image carousel with custom pagination
 */
export const ImageSwiper = ({ images }: ImageSwiperProps) => {
  return (
    <View style={styles.container}>
      <Swiper
        style={styles.wrapper}
        dot={<View style={styles.dot} />}
        activeDot={<View style={styles.activeDot} />}
        paginationStyle={styles.pagination}
        loop={false}
        showsButtons={false}
        showsPagination={true}>
        {images.map((image, index) => (
          <View key={index} style={styles.slide}>
            <Image source={{ uri: image }} style={styles.image} resizeMode="contain" />
          </View>
        ))}
      </Swiper>
    </View>
  );
};

// Styles for component layout and appearance
const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginHorizontal: MARGINS.horizontal,
    marginTop: 12,
    marginBottom: 40,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  wrapper: {},
  slide: {
    flex: 1,
  },
  image: {
    width: "100%",
    height: "100%",
  },
  pagination: {
    bottom: -20,
  },
  dot: {
    backgroundColor: COLORS.background,
    width: 8,
    height: 8,
    borderRadius: 5,
    marginHorizontal: 3,
    shadowColor: "rgba(0, 0, 0, 0.5)",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  activeDot: {
    backgroundColor: COLORS.primary,
    width: 10,
    height: 10,
    borderRadius: 5,
    marginHorizontal: 3,
    shadowColor: "rgba(0, 0, 0, 0.5)",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
