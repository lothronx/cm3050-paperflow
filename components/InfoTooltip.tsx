import { useState } from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Tooltip from "react-native-walkthrough-tooltip";
import { Ionicons } from "@expo/vector-icons";
import { COLORS } from "@/constants/Colors";


interface InfoTooltipProps {
  content: string;
}

export const InfoTooltip = ({ content }: InfoTooltipProps) => {
  const [showTooltip, setShowTooltip] = useState(false);

  return (
    <Tooltip
      isVisible={showTooltip}
      content={<Text>{content}</Text>}
      placement="top"
      onClose={() => setShowTooltip(false)}>
      <TouchableOpacity onPress={() => setShowTooltip(true)}>
        <View style={styles.helpIcon}>
          <Ionicons name="help-circle" size={16} color={COLORS.textSecondary} />
        </View>
      </TouchableOpacity>
    </Tooltip>
  );
};

const styles = StyleSheet.create({
  helpIcon: {
    marginLeft: 4,
  },
});
