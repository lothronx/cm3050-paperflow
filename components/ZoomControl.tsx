import { Pressable, StyleSheet } from "react-native";
import { MaterialIcons } from "@expo/vector-icons";
import { COLORS } from "@/constants/Colors";

interface ZoomControlProps {
  isZoomedIn: boolean;
  onToggle: () => void;
}

export const ZoomControl = ({ isZoomedIn, onToggle }: ZoomControlProps) => (
  <Pressable onPress={onToggle} style={styles.zoomButton} testID="zoom-button">
    <MaterialIcons
      name={isZoomedIn ? "zoom-in-map" : "zoom-out-map"}
      size={18}
      color={COLORS.primary}
    />
  </Pressable>
);

const styles = StyleSheet.create({
  zoomButton: {
    position: "absolute",
    bottom: 10,
    left: "50%",
    transform: [{ translateX: -17 }],
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 999,
    zIndex: 1,
    backgroundColor: COLORS.background,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
});
