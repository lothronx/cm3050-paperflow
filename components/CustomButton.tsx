import { StyleSheet, TouchableOpacity, View } from "react-native";
import { COLORS } from "@/constants/Colors";
import { Text } from "@/components/Text";

interface CustomButtonProps {
  text: string;
  onPress: () => void;
  variant?: "solid" | "outline";
  icon?: React.ReactNode;
}

export const CustomButton = ({ text, onPress, variant = "solid", icon }: CustomButtonProps) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      activeOpacity={0.8}
      style={[styles.button, variant === "outline" && styles.outlineButton]}>
      <View style={styles.buttonContent}>
        {icon && <View style={styles.iconContainer}>{icon}</View>}
        <Text style={[styles.buttonText, variant === "outline" && styles.outlineButtonText]}>
          {text}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    paddingVertical: 16,
    paddingHorizontal: 32,
    borderRadius: 12,
    alignItems: "center",
    marginHorizontal: 16,
    backgroundColor: COLORS.primary,
    shadowColor: "rgba(0, 0, 0, 0.5)",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  outlineButton: {
    backgroundColor: COLORS.background,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: COLORS.border,
  },
  buttonText: {
    color: COLORS.background,
    fontSize: 18,
    fontFamily: "Montserrat-Medium",
    textShadowColor: "rgba(0, 0, 0, 0.25)",
    textShadowOffset: {
      width: 0,
      height: 2,
    },
    textShadowRadius: 3.84,
  },
  outlineButtonText: {
    color: COLORS.primary,
    textShadowColor: "rgba(0, 0, 0, 0.1)",
    textShadowOffset: {
      width: 0,
      height: 2,
    },
    textShadowRadius: 3.84,
  },
  buttonContent: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  iconContainer: {
    marginRight: 8,
  },
});
