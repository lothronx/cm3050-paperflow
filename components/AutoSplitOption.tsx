import { StyleSheet, View, Switch } from "react-native";
import { COLORS } from "@/constants/Colors";
import { Text } from "@/components/Text";
import { InfoTooltip } from "@/components/InfoTooltip";
import { MARGINS } from "@/constants/Margins";
interface AutoSplitOptionProps {
  title: string;
  tooltip?: string;
  defaultValue: boolean;
  onValueChange: (value: boolean) => void;
}

export const AutoSplitOption = ({ title, tooltip, defaultValue, onValueChange }: AutoSplitOptionProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
        {tooltip && <InfoTooltip content={tooltip} />}
      </View>
      <Switch
        value={defaultValue}
        onValueChange={onValueChange}
        trackColor={{ false: COLORS.border, true: COLORS.primary }}
        ios_backgroundColor={COLORS.border}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingVertical: 12,
    paddingHorizontal: MARGINS.horizontal,
  },
  titleContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  title: {
    color: COLORS.text,
    marginRight: 4,
  },
});
