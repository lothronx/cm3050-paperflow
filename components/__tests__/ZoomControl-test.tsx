import { render, fireEvent } from "@testing-library/react-native";
import { ZoomControl } from "@/components/ZoomControl";

// Mock the MaterialIcons component
jest.mock("@expo/vector-icons", () => ({
  MaterialIcons: "MaterialIcons",
}));

describe("ZoomControl", () => {
  const mockOnToggle = jest.fn();

  it("should match snapshot", () => {
    const tree = render(<ZoomControl isZoomedIn={false} onToggle={mockOnToggle} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onToggle when pressed", () => {
    const { getByTestId } = render(<ZoomControl isZoomedIn={false} onToggle={mockOnToggle} />);
    
    const zoomButton = getByTestId("zoom-button");
    fireEvent.press(zoomButton);
    
    expect(mockOnToggle).toHaveBeenCalledTimes(1);
  });

  it("uses correct icon name based on isZoomedIn prop", () => {
    const { UNSAFE_queryByProps, rerender } = render(
      <ZoomControl isZoomedIn={false} onToggle={mockOnToggle} />
    );

    // Check icon when not zoomed in
    let icon = UNSAFE_queryByProps({ name: "zoom-out-map" });
    expect(icon).toBeTruthy();

    // Rerender with isZoomedIn=true and check icon
    rerender(<ZoomControl isZoomedIn={true} onToggle={mockOnToggle} />);
    icon = UNSAFE_queryByProps({ name: "zoom-in-map" });
    expect(icon).toBeTruthy();
  });
});
