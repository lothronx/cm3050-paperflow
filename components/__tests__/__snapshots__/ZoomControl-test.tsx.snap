// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomControl should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 999,
      "bottom": 10,
      "elevation": 5,
      "left": "50%",
      "paddingHorizontal": 8,
      "paddingVertical": 6,
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "transform": [
        {
          "translateX": -17,
        },
      ],
      "zIndex": 1,
    }
  }
  testID="zoom-button"
>
  <MaterialIcons
    color="#6A7BFF"
    name="zoom-out-map"
    size={18}
  />
</View>
`;
